### PR TITLE
[BIOMAGE-1226] Implement the backend for download of the raw Seurat object

### PR DESCRIPTION
**Pull request by StefanBabukov**
# Background
#### Link to issue 
https://biomage.atlassian.net/secure/RapidBoard.jspa?rapidView=1&projectKey=BIOMAGE&modal=detail&selectedIssue=BIOMAGE-1226
#### Link to staging deployment URL 
https://ui-grasp6-ui420-api196.scp-staging.biomage.net/
#### Anything else the reviewers should know about the changes here

# Changes
### Code changes

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Testing
- [ ] Unit tests written
- [ ] Tested locally with Inframock (with latest production data downloaded with biomage experiment pull)
- [ ] Deployed to staging

To set up easy local testing with inframock, follow the instructions here: https://github.com/biomage-ltd/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/biomage-ltd/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [ ] Relevant Github READMEs updated
- [ ] Relevant wiki pages created/updated

### Approvers
- [ ] Approved by a member of the core engineering team
- [ ] (UX changes) Approved by vickymorrison (this is her username, tag her if you need approval)

### Just before merging:
- [ ] After the PR is approved, the `unstage` script in here: https://github.com/biomage-ltd/biomage-utils is executed. This script cleans up your deployment to staging

### Optional
- [ ] Photo of a cute animal attached to this PR

### CODE REVIEW

**Comment by ivababukova** [comment]
@StefanBabukov I see this has been opened for a while. How about you call someone today and go through the code together? Let's get it reviewed so we can merge it soon.

**Comment by StefanBabukov** [comment]
@aerlaut has been reviewing it

File: src/components/data-management/ProjectDetails.jsx [hunk 1]
```diff
@@ -636,20 +636,38 @@ const ProjectDetails = ({ width, height }) => {
   const pipelineHasRun = (experimentId) => (
     experiments[experimentId]?.meta?.pipeline?.status === pipelineStatus.SUCCEEDED
   );
+  const gem2sHasRun = (experimentId) => (
+    experiments[experimentId]?.meta?.gem2s?.status === pipelineStatus.SUCCEEDED
+  );
```
Comment (aerlaut) [inline src/components/data-management/ProjectDetails.jsx]: these checks are updated, check PR #422 

```
experiments[experimentId]?.meta?.backendStatus?.pipeline?.status === pipelineStatus.SUCCEEDED
experiments[experimentId]?.meta?.backendStatus?.gem2s?.status === pipelineStatus.SUCCEEDED
```

**Review by aerlaut — APPROVED** [review]
LGTM

**Pull request merged**